### PR TITLE
issue 64

### PR DESCRIPTION
**Pull request by ltkum**
#64 
TO DO LIST
- [x] check profile max length 
- [x] change the name of keep_points to "distinct_points"

as discussed with pakb, I created a metadata parameter which, for now, does nothing. The goal will be to have a smarter profile json, with some metadata informations given and less redundancy in the data. I will try to add the metadata approach to the PR too

### CODE REVIEW

**Comment by pakb** [comment]
image
bad boy! baaad!

now you can fix your code so that unit tests pass 😉

**Comment by pakb** [comment]
wait, I have to adapt them a little bit with your new code

**Comment by ltkum** [comment]
I'll wait, I'll wait ^^

File: alti/lib/profile_helpers.py [hunk 1]
```diff
@@ -82,9 +100,28 @@ def _create_profile(coordinates, z_values, output_to_json):
     return profile
 
 
-def _smart_fill(coordinates, nb_points):
+def _prepare_number_of_points_max_per_segment(coordinates, nb_point_total):
+
+    nb_points_segments = list()
+    distances = []
+    distances_squareform = squareform(pdist(coordinates))
```
Comment (loicgasser) [inline alti/lib/profile_helpers.py:107]: Hey I don't mean to be intrusive, I am just interested in the sampling method for the profile.

That seems like a lot of extra unnecessary work no? you are computing n^2 distances when you only need the distance between each point (n-1). Wouldn't "Pythagoras Theorem" without scipy be a better alternative? I think you could avoid a bunch of loops and keep the code much tighter. Something like:

```python
import math
def square_dist(a, b):
    return math.sqrt(math.pow(a[0] - b[0], 2) + math.pow(a[1] - b[1], 2))
distances = []
total_disance = 0
# If python 2
for i in xrange(0, len(coordinates) - 1):
    dist = square_dist(coordinates[i], coordinates[i+1])
    distances.append(dist)
    total_distance += dist
```

**Comment by procrastinatio** [comment]
`make autolint` is your new best friend...

File: alti/lib/profile_helpers.py (hunk 1 above)
Comment (ltkum) [reply alti/lib/profile_helpers.py:107]: That's true. I simply reused the existing method, thinking there was a reason why it was done this way.

**Comment by pakb** [comment]
Could you run a little profiling on the code and optimize a little bit? I've run a yandextank on an INT instance, and it found that we have lost about 25% of performance with this PR. With the `master` code I can have 80ish request per second on the INT instance, with this branch it's down to about 50. 
I'm using the default params, here's the request I use as my ammo `/rest/services/profile.json?offset=0` with `{"type":"LineString","coordinates":[[2629499.8,1170351.9],[2635488.4,1173402.0]]}` as its body.

I don't mind we could loose some perfo with `smart_fill` and `keep_points` as they will be used only by geoadmin, but I have some concern if we degrade default performances.

**Pull request merged**